bitrisescript/src/bitrisescript/task.py

```diff
@@ -1,10 +1,26 @@
 import os
+from copy import deepcopy
 from typing import Any
 
 from scriptworker_client.exceptions import TaskVerificationError
 from scriptworker_client.utils import get_single_item_from_sequence
 
 
+def _deep_merge_dict(base: dict, ref: dict) -> dict:
```

Comment:
You might just want to copy/paste this function:
https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/util/templates.py#L40

It's pretty battle tested.